core/commands/id.go

```diff
@@ -93,30 +94,36 @@ EXAMPLE:
 
 		// TODO handle offline mode with polymorphism instead of conditionals
 		if !node.OnlineMode() {
-			res.SetError(errors.New(offlineIdErrorMessage), cmds.ErrClient)
+			res.SetError(errors.New(offlineIdErrorMessage), cmdsutil.ErrClient)
 			return
 		}
 
 		p, err := node.Routing.FindPeer(req.Context(), id)
 		if err == kb.ErrLookupFailure {
-			res.SetError(errors.New(offlineIdErrorMessage), cmds.ErrClient)
+			res.SetError(errors.New(offlineIdErrorMessage), cmdsutil.ErrClient)
 			return
 		}
 		if err != nil {
-			res.SetError(err, cmds.ErrNormal)
+			res.SetError(err, cmdsutil.ErrNormal)
 			return
 		}
 
 		output, err := printPeer(node.Peerstore, p.ID)
 		if err != nil {
-			res.SetError(err, cmds.ErrNormal)
+			res.SetError(err, cmdsutil.ErrNormal)
 			return
 		}
 		res.SetOutput(output)
 	},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			val, ok := res.Output().(*IdOutput)
+			ch, ok := res.Output().(chan interface{})
+			if !ok {
+				log.Debugf("received type %T", res.Output())
+				return nil, u.ErrCast()
```

Comment:
maybe we should have a new error thing that prints out the expected and actual types. Instead of the ugly `ErrCast` nonsense (no need to do this now though)